### ISSUE TITLE
Great work, question about the quick start code

### ISSUE THREAD

**Issue by yanghu819** [xichen-fy/Fira#1]
In the quick start demo, I found these code are not running, which not use the subgrad at all, what could be my mistakes, I just run your code and did no adaptation.
                if "rank" in group:
                    # Norm-Based Scaling
                    subgrad = state["projector"].project_back(grad)
                    norm_dim = 0 if norm_grad.shape[0] < norm_grad.shape[1] else 1
                    scaling_factor = (
                        torch.norm(norm_grad, dim=norm_dim) /
                        (torch.norm(grad, dim=norm_dim) + 1e-8)
                    )
                    if norm_dim == 1:
                        scaling_factor = scaling_factor.unsqueeze(1)
                    scaling_grad = (p.grad - subgrad) * scaling_factor

                    # Norm-Growth Limiter
                    if "scaling_grad" in state:
                        scaling_grad_norm = torch.norm(scaling_grad)
                        limiter = max(
                                scaling_grad_norm / 
                                (state["scaling_grad"] + 1e-8),
                                1.01,
                            ) / 1.01
                        scaling_grad = scaling_grad / limiter
                        state["scaling_grad"] = scaling_grad_norm / limiter
                    else:
                        state["scaling_grad"] = torch.norm(scaling_grad)
                    
                    norm_grad = state["projector"].project_back(norm_grad) + scaling_grad

**Comment by xichen-fy** [comment]
Thank you for your question, I will check and reply soon.

yanghu819 ***@***.***>于2024年10月20日 周日19:08写道：

**Comment by fy817** [comment]
Thank you for your inquiry. I have verified that gradient projection is not enabled because the parameter `target_modules_list` in the function `divide_params` is incorrect:
```python
from fira import FiraAdamW, divide_params
# param_groups = divide_params(model, target_modules_list=["Linear"], rank=8)
param_groups = divide_params(model, target_modules_list=["fc"], rank=8)
optimizer = FiraAdamW(param_groups, lr=learning_rate)
```
This parameter should correspond to the `module_name` used in the model's code, such as substring `fc` in the following example, not to the type of module like `Linear`:
```python
class SimpleNN(nn.Module):
    def __init__(self):
        super(SimpleNN, self).__init__()
        self.fc1 = nn.Linear(28*28, 128)  # First layer
        self.fc2 = nn.Linear(128, 10)     # Output layer
```
I appreciate your question and will update the README file soon to clarify this.

**Comment by yanghu819** [comment]
Great!
